### PR TITLE
feat user profile

### PR DESCRIPTION
**Pull request by deepak29-git**
user can see their profile on profile tab 
image

### CODE REVIEW

File: src/Pages/Profile/Profile.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { getAllUser, getUser } from "../../all-api/user-api";
+import { EditProfileModal } from "../../Components/EditProfileModal/EditProfileModal";
+import { Sidebar } from "../../Components/Sidebar/Sidebar";
+import { useUser } from "../../Context/user-context";
+
+export const Profile = () => {
+  const { userState, userDispatch } = useUser();
+  const { users } = userState;
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  // const [userId,setUserId]=useState()
```
Comment (tushargoyal812) [inline src/Pages/Profile/Profile.jsx:12]: remove comments

File: src/all-api/post-api.js [hunk 2]
```diff
@@ -26,33 +26,34 @@ const createPostApi = async (postDispatch, post) => {
 };
 
 const editPostApi = async (postDispatch, _id, postData) => {
-  console.log(postDispatch,"dispatch", _id,"id", postData,"input")
+  console.log(postDispatch, "dispatch", _id, "id", postData, "input");
```
Comment (tushargoyal812) [inline src/all-api/post-api.js:29]: remove console

File: src/all-api/user-api.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const { data } = await axios.get(`/api/users/${id}`);
+    userDispatch({ type: "GET_USER", payload: data.user });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const editProfileApi = async (userData, userDispatch) => {
+  const { _id, firstName, lastName, userName, bio } = userData;
+  console.log(firstName, "from frontend");
```
Comment (tushargoyal812) [inline src/all-api/user-api.js:24]: remove console

File: src/all-api/user-api.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      data: {
+        firstName: firstName,
+        lastName: lastName,
+        userName: userName,
+        bio: bio,
+      },
+      headers: { authorization: getToken() },
+    });
+    console.log(data.user, "from edit api");
+    // userDispatch({ type: "EDIT_USER", payload: data.users});
```
Comment (tushargoyal812) [inline src/all-api/user-api.js:38]: remove console and comments

File: src/backend/db/users.js [hunk 5]
```diff
@@ -6,17 +6,21 @@ import { formatDate } from "../utils/authUtils";
...
 export const users = [
+  // {
+  //   _id: uuid(),
+  //   firstName: "Adarsh",
+  //   lastName: "Balika",
+  //   username: "adarshbalika",
+  //   password: "adarshBalika123",
+  //   createdAt: formatDate(),
+  //   updatedAt: formatDate(),
+  // },
```
Comment (tushargoyal812) [inline src/backend/db/users.js:17]: remove comments

**Review by shivanigangadharan — APPROVED** [review]
Apart from changes suggested by Tushar, everything looks good. Can merge after removing consoles and comments.

File: src/Pages/Profile/Profile.jsx (hunk 1 above)
Comment (deepak29-git) [reply src/Pages/Profile/Profile.jsx:12]: ok

File: src/all-api/user-api.js (hunk 3 above)
Comment (deepak29-git) [reply src/all-api/user-api.js:24]: ok

File: src/backend/db/users.js (hunk 5 above)
Comment (deepak29-git) [reply src/backend/db/users.js:17]: ok

File: src/all-api/user-api.js (hunk 4 above)
Comment (deepak29-git) [reply src/all-api/user-api.js:38]: ok

File: src/all-api/post-api.js (hunk 2 above)
Comment (deepak29-git) [reply src/all-api/post-api.js:29]: ok

**Pull request merged**